forte/data/ontology/top.py

```diff
@@ -895,17 +896,82 @@ def __eq__(self, other):
         return self.image_payload_idx == other.image_payload_idx
 
 
-class Grids(Entry):
+class Region(ImageAnnotation):
     """
-    Regular grids with a grid configuration.
+    A region class associated with an image payload.
 
     Args:
-        pack: The container that this grids will be added to.
-        height: the number of grid cell per column, the unit is one grid cell.
-        width: the number of grid cell per row, the unit is one grid cell.
-        image_payload_idx: the index of the image payload in the DataPack's
-            image payload list.
-            If it's not set,
+        pack: the container that this ``Region`` will be added to.
+        image_payload_idx: the index of the image payload. If it's not set,
+            it defaults to 0 which meaning it will load the first image payload.
+    """
+
+    def __init__(self, pack: PackType, image_payload_idx: int = 0):
+        super().__init__(pack, image_payload_idx)
+        if image_payload_idx is None:
+            self._image_payload_idx = 0
+        else:
+            self._image_payload_idx = image_payload_idx
+
+    def compute_iou(self, other) -> float:
+        intersection = np.sum(np.logical_and(self.image, other.image))
+        union = np.sum(np.logical_or(self.image, other.image))
+        return intersection / union
+
+
+class Box(Region):
+    # pylint: disable=too-many-public-methods
+    """
+    A box class with a reference point which is the box center and a box
+    configuration.
+
+    Box of a certain size (height, width), as a regular shape, can be located
+    only when it has a reference point (box center here).
+
+    There are several use cases for a box:
+        1. When we use a box standalone, we need the box center to be set.The offset
+        between the box center and the grid cell center is not used.
+        2. When we represent a ground truth box, the box center and shape are
+        given. If we want to compute loss, its grid center is required to
+        compute the offset between the box center and the
+        grid cell center.
+        3. When we predict a box, we will have the predicted box shape (height,
+        width) and the offset between the box center and the grid cell center,
+        then we can compute the box center.
+
+    For example, in the object detection task, dataset label contains a ground
+    truth box (box shape and box center).
+    The inference pipeline is that given a grid cell, we make a prediction of a
+    bounding box (box shape and box offset from the grid cell center).
+        1. If we want to locate the predicted box, we compute the box center
+        based on the second use case.
+        2. If we want to compute the loss, we need values for the center, shape
+        and offset for both boxes. And we need to compute the offset between
+        the box center and the grid cell center based on the third use case.
```

Comment:
Can you explain the examples using the code snippets (how a box should be created?)

Reply:
Added code